digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala

```diff
@@ -1236,4 +1236,70 @@ class ProjectServiceSpec extends FunSuite with Matchers with BeforeAndAfter {
       response.get should be ("TIE 5 OSA 203 on jo olemassa projektin alkupäivänä 01.01.1901, tarkista tiedot")
     }
   }
+
+  ignore("splitting road address and saving with different ROADTYPE on transfer + new creates one new commonHistoryId for the segment") {
+    val road = 5L
+    val roadPart = 205L
+    val origStartM = 1024L
+    val origEndM = 1547L
+    val origStartD = Some(DateTime.now().minusYears(10))
+    val linkId = 1049L
+    val endM = 520.387
+    val suravageLinkId = 5774839L
+    val user = Some("user")
+    val project = RoadAddressProject(-1L, Sent2TR, "split", user.get, DateTime.now(), user.get,
+      DateTime.now().plusMonths(2), DateTime.now(), "", Seq(), None, None)
+    val roadAddress = RoadAddress(1L, road, roadPart, PublicRoad, Track.Combined, Continuous, origStartM, origEndM, origStartD,
+      None, None, 1L, linkId, 0.0, endM, SideCode.TowardsDigitizing, 86400L, (None, None), false, Seq(Point(1024.0, 0.0), Point(1025.0, 1544.386)),
+      LinkGeomSource.NormalLinkInterface, 8L, NoTermination, 123456)
+    val transferAndNew = Seq(ProjectLink(2L, road, roadPart, Track.Combined, Continuous, 1028, 1128, Some(DateTime.now()), None, user,
+      2L, suravageLinkId, 0.0, 99.384, SideCode.TowardsDigitizing, (None, None), false, Seq(Point(1024.0, 0.0), Point(1024.0, 99.384)),
+      -1L, LinkStatus.Transfer, MunicipalityStreetRoad, LinkGeomSource.SuravageLinkInterface, 99.384, 1L, 8L, false, Some(linkId), 748800L),
+      ProjectLink(3L, road, roadPart, Track.Combined, Continuous, 1128, 1205, Some(DateTime.now()), None, user,
+        3L, suravageLinkId, 99.384, 176.495, SideCode.TowardsDigitizing, (None, None), false, Seq(Point(1024.0, 99.384), Point(1101.111, 99.384)),
+        -1L, LinkStatus.New, MunicipalityStreetRoad, LinkGeomSource.SuravageLinkInterface, 77.111, 1L, 8L, false, Some(linkId), 748800L),
+      ProjectLink(4L, road, roadPart, Track.Combined, Continuous, origStartM + 100L, origEndM, Some(DateTime.now()), None, user,
+        4L, linkId, 99.384, endM, SideCode.TowardsDigitizing, (None, None), false, Seq(Point(1024.0, 99.384), Point(1025.0, 1544.386)),
+        -1L, LinkStatus.Terminated, MunicipalityStreetRoad, LinkGeomSource.NormalLinkInterface, endM - 99.384, 1L, 8L, false, Some(suravageLinkId), 748800L))
+    val result = projectService.createSplitRoadAddress(roadAddress, transferAndNew, project)
+    result should have size(4)
+    result.count(_.terminated == TerminationCode.Termination) should be (1)
+    result.count(_.startDate == roadAddress.startDate) should be (2)
+    result.count(_.startDate.get == project.startDate) should be (2)
+    result.count(_.endDate.isEmpty) should be (2)
+    result.filter(_.terminated == TerminationCode.NoTermination).foreach(r => r.commonHistoryId should be (-1000))
+  }
+
+  ignore("splitting road address and saving with different TRACK on transfer + new creates one new commonHistoryId for the segment") {
```

Comment:
why are these tests ignored?

Reply:
those two ignored tests should be uncommented in next sprint where the validations from user changes will need to be checked. I will remove them since we dont need them yet.